test/unit/services/DeveloperServiceSpec.scala

```diff
@@ -81,31 +79,72 @@ class DeveloperServiceSpec extends UnitSpec with MockitoSugar {
           "application3", None, collaborators, DateTime.now(), ApplicationState()))
 
       given(testDeveloperService.applicationConnector.fetchAllApplicationsBySubscription(mEq("subscription"))(any[HeaderCarrier])).willReturn(Future.successful(filteredApplications))
-      given(testDeveloperService.developerConnector.fetchAll()(any[HeaderCarrier])).willReturn(Future.successful(users))
-      given(testDeveloperService.apiDefinitionConnector.fetchAll()(any[HeaderCarrier])).willReturn(Seq.empty[APIDefinition])
+      given(testDeveloperService.applicationConnector.fetchAllApplicationsBySubscription(mEq("subscription"))(any[HeaderCarrier])).willReturn(Future.successful(filteredApplications))
       val result = await(testDeveloperService.fetchApplications(Value("subscription"))(new HeaderCarrier()))
       result shouldBe filteredApplications
     }
 
+    "list filtered developers when OneOrMoreSubscriptions filtering is provided" in new Setup {
+      val users = Seq(user("Bob", false), user("Brian"), user("Sheila"))
+
+      val noSubscriptions = Seq(
+        ApplicationResponse(UUID.randomUUID(),
+          "application1", None, collaborators, DateTime.now(), ApplicationState()),
+        ApplicationResponse(UUID.randomUUID(),
+          "application3", None, collaborators, DateTime.now(), ApplicationState()))
+
+      val subscriptions = Seq(
+        ApplicationResponse(UUID.randomUUID(),
+          "application2", None, collaborators, DateTime.now(), ApplicationState()),
+        ApplicationResponse(UUID.randomUUID(),
+          "application4", None, collaborators, DateTime.now(), ApplicationState()))
+
+      val allApps = noSubscriptions ++ subscriptions
+      given(testDeveloperService.applicationConnector.fetchAllApplications()(any[HeaderCarrier])).willReturn(Future.successful(allApps))
+      given(testDeveloperService.applicationConnector.fetchAllApplicationsWithNoSubscriptions()(any[HeaderCarrier])).willReturn(Future.successful(noSubscriptions))
+      val result = await(testDeveloperService.fetchApplications(OneOrMoreSubscriptions)(new HeaderCarrier()))
+      result shouldBe subscriptions
+    }
+
+    "list filtered developers when OneOrMoreApplications filtering is provided" in new Setup {
+      val users = Seq(user("Bob", false), user("Brian"), user("Sheila"))
```

Comment:
The setup in this test is identical to the previous one - can it be extracted to a common setup?